### DISCUSSION TITLE
Bounding a parameter and a variable

### DISCUSSION THREAD

**Discussion by GabrielWellander** [alphaville/optimization-engine#364] [Q&A] [answered]
Hello,
I am using OpEn to make an MPC for a underwater vehicle and saw that the velocity in certain degrees became too big.
Is there a way to use something similar as .with_constraints() but for a parameter and not a variable?
Best Regards,
GW

**Comment by alphaville** [comment, marked as answer]
Sorry for the late reply. Can you give an example of what you have in mind? If you want to impose a constraint of the form $p \in P$, this can be done outside the solver since $p$ is not a decision variable. Constraints of the form $F_1(u, p) \in C$ and $F_2(u, p) = 0$ are supported. Constraints of the form $u \in U(p)$ are currently not supported, but it would be a good idea to support them somehow. This can be easily done in Rust, but through the code generator it will be more tricky.
